src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,323 @@
+     */
+    private EmptyLineWrappingInBlockOption topSeparator =
+        EmptyLineWrappingInBlockOption.EMPTY_LINE_ALLOWED;
+
+    /**
+     * Specify the policy of empty line at the bottom top of the code block.
+     */
+    private EmptyLineWrappingInBlockOption bottomSeparator =
+        EmptyLineWrappingInBlockOption.EMPTY_LINE_ALLOWED;
+
+    /**
+     * Setter to specify the policy of empty line on top of the code block.
+     *
+     * @param option string to decode option from
+     */
+    public void setTopSeparator(String option) {
+        topSeparator = EmptyLineWrappingInBlockOption.valueOf(
+            option.trim().toUpperCase(Locale.ENGLISH)
+        );
+    }
+
+    /**
+     * Setter to specify the policy of empty line at the bottom top of the code block.
+     *
+     * @param option string to decode option from
+     */
+    public void setBottomSeparator(String option) {
+        bottomSeparator = EmptyLineWrappingInBlockOption.valueOf(
+            option.trim().toUpperCase(Locale.ENGLISH)
+        );
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_TRY,
+            TokenTypes.LITERAL_IF,
+            TokenTypes.LITERAL_ELSE,
+            TokenTypes.LITERAL_SWITCH,
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.LITERAL_FOR,
+            TokenTypes.LITERAL_WHILE,
+            TokenTypes.LITERAL_DO,
+            TokenTypes.STATIC_INIT,
+            TokenTypes.INSTANCE_INIT,
+            TokenTypes.ANNOTATION_DEF,
+            TokenTypes.ENUM_DEF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (hasEmptyLineAfter(ast)) {
+            if (bottomSeparator == EmptyLineWrappingInBlockOption.NO_EMPTY_LINE) {
+                log(ast, MSG_NO_EMPTY_LINE_AFTER);
+            }
+        }
+        else if (bottomSeparator == EmptyLineWrappingInBlockOption.EMPTY_LINE) {
+            log(ast, MSG_EMPTY_LINE_AFTER);
+        }
+        if (hasEmptyLineBefore(ast)) {
+            if (topSeparator == EmptyLineWrappingInBlockOption.NO_EMPTY_LINE) {
+                log(ast, MSG_NO_EMPTY_LINE_BEFORE);
+            }
+        }
+        else if (topSeparator == EmptyLineWrappingInBlockOption.EMPTY_LINE) {
+            log(ast, MSG_EMPTY_LINE_BEFORE);
+        }
+    }
+
+    /**
+     * Checks whether an empty/blank line is present before the given block.
+     *
+     * @param ast the current node
+     * @return true if an empty/blank line is present before the given block.
+     */
+    private boolean hasEmptyLineBefore(DetailAST ast) {
```

Comment:
You actually should find the closing curly and look at the previous line.

Reply:
But checking the line before the block node would do the same thing right ?